packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1568,53 +1616,28 @@ export class TransactionController extends BaseController<
     return remoteGasUsed !== localGasUsed;
   }
 
-  private async requestApproval(
-    txMeta: TransactionMeta,
-    { shouldShowRequest } = { shouldShowRequest: true },
-  ) {
+  private async requestApproval(txMeta: TransactionMeta): Promise<AddResult> {
     const id = this.getApprovalId(txMeta);
     const { origin } = txMeta;
     const type = ApprovalType.Transaction;
     const requestData = { txId: txMeta.id };
 
-    try {
-      await this.messagingSystem.call(
-        'ApprovalController:addRequest',
-        {
-          id,
-          origin: origin || ORIGIN_METAMASK,
-          type,
-          requestData,
-        },
-        shouldShowRequest,
```

Comment:
Assuming that we removed the `shouldShowRequest` because this controller is used on mobile... but eventually when we merge both transaction controllers (extension + mobile) we will have to re-add this. Right?

Reply:
Not sure I follow, we still provide the argument as `true` to the `ApprovalController`, we just removed the internal argument as all our use cases want the approval shown.